dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java

```diff
@@ -0,0 +1,519 @@
+
+            Logger.debug(this, "Looking for the action: " + actionId);
+            action = this.workflowAPI.findAction
+                    (actionId, user);
+        } catch (DotDataException | DotSecurityException e) {
+
+            Logger.error(this, e.getMessage(), e);
+        }
+
+        if (null != action) {
+
+            try {
+
+                Logger.debug(this, "Deleting the action: " + actionId);
+                this.workflowAPI.deleteAction(action);
+            } catch (DotDataException | AlreadyExistException e) {
+
+                Logger.error(this, e.getMessage(), e);
+                throw new DotWorkflowException(e.getMessage(), e);
+            }
+        } else {
+
+            throw new DoesNotExistException("Workflow-does-not-exists-action");
+        }
+    } // deleteAction.
+
+        /**
+         * Deletes the action which is part of the step, but the action still being part of the scheme.
+         * @param actionId String action id
+         * @param stepId   String step   id
+         * @param user     User   the user that makes the request
+         * @return WorkflowStep
+         */
+    @WrapInTransaction
+    public WorkflowStep deleteAction(final String actionId,
+                                     final String stepId,
+                                     final User user) {
+
+        WorkflowAction action = null;
+        WorkflowStep step     = null;
+
+        try {
+
+            Logger.debug(this, "Looking for the actionId: " + actionId);
+            action =
+                    this.workflowAPI.findAction(actionId, user);
+
+            Logger.debug(this, "Looking for the stepId: " + stepId);
+            step =
+                    this.workflowAPI.findStep(stepId);
+
+            if (null == action) {
+                throw new DoesNotExistException("Workflow-does-not-exists-action");
+            }
+
+            if (null == step) {
+                throw new DoesNotExistException("Workflow-does-not-exists-step");
+            }
+
+            Logger.debug(this, "Deleting the action: " + actionId
+                    + " for the stepId: " + stepId);
+
+            this.workflowAPI.deleteAction(action, step);
+        } catch (DotDataException | DotSecurityException | AlreadyExistException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotWorkflowException(e.getMessage(), e);
+        }
+
+        return step;
+    } // deleteAction.
+
+    /**
+     * Finds the action associated to the stepId
+     * @param stepId String
+     * @param user   User
+     * @return List of WorkflowAction
+     */
+    public List<WorkflowAction> findActions(final String stepId, final User user) {
+
+        WorkflowStep workflowStep    = null;
+        List<WorkflowAction> actions = null;
+
+        try {
+
+            Logger.debug(this, "Looking for the stepId: " + stepId);
+            workflowStep = this.workflowAPI.findStep(stepId);
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)